save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt

```diff
@@ -34,19 +39,34 @@ class TestService(private val configProperties: ConfigProperties) {
     @Autowired
     private lateinit var testExecutionRepository: TestExecutionRepository
 
+    @Autowired
+    private lateinit var testSuiteRepository: TestSuiteRepository
+
     /**
      * @param tests
+     * @return list tests id's
      */
-    fun saveTests(tests: List<Test>) {
-        testRepository.saveAll(tests)
+    fun saveTests(tests: List<TestDto>): List<Long> {
+        val testsId: MutableCollection<Long> = mutableListOf()
+        tests.map { testDto ->
+            testRepository.findByHash(testDto.hash)?.let { testsId.add(it.id!!) } ?: run {
+                testSuiteRepository.findById(testDto.testSuiteId).ifPresent { testSuite ->
+                    Test(testDto.hash, testDto.filePath, LocalDateTime.now(), testSuite).run {
```

Comment:
This is the same thing that I talked about: you are making an additional query to get test suite, even though it's not updated and we only need it's id in `test` table. Please check generated SQL in logs, whether hibernate optimizes this somehow or just makes all the queries.